katharsis-client/src/main/java/io/katharsis/client/KatharsisClient.java

```diff
@@ -0,0 +1,226 @@
+package io.katharsis.client;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.squareup.okhttp.OkHttpClient;
+
+import io.katharsis.client.internal.BaseResponseDeserializer;
+import io.katharsis.client.internal.ErrorResponseDeserializer;
+import io.katharsis.client.internal.RelationshipRepositoryStubImpl;
+import io.katharsis.client.internal.RequestUrlBuilder;
+import io.katharsis.client.internal.ResourceRepositoryStubImpl;
+import io.katharsis.errorhandling.ErrorResponse;
+import io.katharsis.errorhandling.mapper.ExceptionMapperLookup;
+import io.katharsis.errorhandling.mapper.ExceptionMapperRegistry;
+import io.katharsis.errorhandling.mapper.ExceptionMapperRegistryBuilder;
+import io.katharsis.jackson.JsonApiModuleBuilder;
+import io.katharsis.module.CoreModule;
+import io.katharsis.module.Module;
+import io.katharsis.module.ModuleRegistry;
+import io.katharsis.repository.RelationshipRepository;
+import io.katharsis.repository.RepositoryInstanceBuilder;
+import io.katharsis.resource.field.ResourceField;
+import io.katharsis.resource.field.ResourceFieldNameTransformer;
+import io.katharsis.resource.information.ResourceInformation;
+import io.katharsis.resource.registry.RegistryEntry;
+import io.katharsis.resource.registry.ResourceLookup;
+import io.katharsis.resource.registry.ResourceRegistry;
+import io.katharsis.resource.registry.repository.DirectResponseRelationshipEntry;
+import io.katharsis.resource.registry.repository.DirectResponseResourceEntry;
+import io.katharsis.resource.registry.repository.ResourceEntry;
+import io.katharsis.resource.registry.repository.ResponseRelationshipEntry;
+import io.katharsis.resource.registry.responseRepository.RelationshipRepositoryAdapter;
+import io.katharsis.resource.registry.responseRepository.ResourceRepositoryAdapter;
+import io.katharsis.response.BaseResponseContext;
+
+/**
+ * Client implementation giving access to JSON API repositories using stubs.
+ */
+public class KatharsisClient {
+
+	private OkHttpClient client = new OkHttpClient();
+	private ObjectMapper objectMapper;
+
+	private ResourceRegistry resourceRegistry;
+
+	private ModuleRegistry moduleRegistry;
+	private RequestUrlBuilder urlBuilder;
+
+	private boolean initialized = false;
+
+	private ExceptionMapperRegistry exceptionMapperRegistry;
+
+	/**
+	 * @param serviceUrl
+	 * @param resourcePackageName
+	 */
+	public KatharsisClient(String serviceUrl, String resourceSearchPackage) {
+		if (serviceUrl.endsWith("/")) {
+			serviceUrl = serviceUrl.substring(0, serviceUrl.length() - 1);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Introduce a new variable instead of reusing the parameter "serviceUrl". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1226)